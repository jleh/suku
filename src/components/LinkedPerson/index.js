import React from 'react';
import { Link } from 'react-router-dom';

import styles from './linkedPerson.css';

import { findPerson, printBirth, printDeath } from '../../util';

export default ({ personRef, persons }) => {
  const person = findPerson(personRef, persons);

  if (!person) {
    return null;
  }

  return (
    <div>
      <Link to={`/person/${person.id}`}>
        {person.name}
      </Link>
      { ' ' }
      <span className={styles.dates}>{printBirth(person)} {printDeath(person)}</span>
    </div>
  );
};
