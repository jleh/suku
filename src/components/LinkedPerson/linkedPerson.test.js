import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import LinkedPerson from '.';

const persons = new Map();

persons.set('H1', {
  handle: 'H1',
  id: '1',
  name: 'Test Person',
  events: {
    birth: '1.1.1900',
    death: '1.1.1950',
  },
});
persons.set('H2', {
  handle: 'H2',
  id: '2',
  name: 'Other Person',
  events: {},
});

const renderComponent = (id) =>
  renderer
    .create(
      <MemoryRouter>
        <LinkedPerson personRef={id} persons={persons} />
      </MemoryRouter>
    )
    .toJSON();

test('Prints person with birth and death dates', () => {
  expect(renderComponent('H1')).toMatchSnapshot();
});

test('Does not fail if person is missing', () => {
  expect(renderComponent('not-fond')).toMatchSnapshot();
});

test('Renders if birth or death is missing', () => {
  expect(renderComponent('H2')).toMatchSnapshot();
});
