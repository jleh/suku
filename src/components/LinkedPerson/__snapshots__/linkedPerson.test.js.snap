// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does not fail if person is missing 1`] = `null`;

exports[`Prints person with birth and death dates 1`] = `
<div>
  <a
    href="/person/1"
    onClick={[Function]}
  >
    Test Person
  </a>
   
  <span
    className="dates"
  >
    * 1.1.1900
     
    â€  1.1.1950
  </span>
</div>
`;

exports[`Renders if birth or death is missing 1`] = `
<div>
  <a
    href="/person/2"
    onClick={[Function]}
  >
    Other Person
  </a>
   
  <span
    className="dates"
  >
     
  </span>
</div>
`;
